### PR TITLE
[13.0][MIG] website_event_filter_selector: Migration to 13.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 13.0

Renamed to `website_event_filter_city`

event-filters

Please @Yajo and @pedrobaeza can you review it?

@Tecnativa TT27591

### CODE REVIEW

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please squash some administrative commits and change readme according the remaining feature (adding the city dropdown). You can rename the module if you think it fits better.

**Comment by victoralmau** [comment]
Readme information is changed although `readme.rst` file is not change because bot will be regenerate it in merge process right?
Too, IMO addon it's need to rename to `website_event_city_filter` what do you think about it?

**Comment by pedrobaeza** [comment]
Following the usual structure, better to be named `website_event_filter_city`, as we don't have to be English order correct, but "drill-through" correct.

**Comment by yajo** [comment]
Could you post some screenshots? runbot doesn't work

File: website_event_filter_selector/controllers/main.py [hunk 1]
```diff
@@ -1,110 +1,59 @@
...
-            )
-            values["countries"][0]["country_id_count"] = sum(
-                int(type_["country_id_count"]) for type_ in values["countries"][1:]
-            )
-            # Patch date count
-            for date in values["dates"]:
-                if date[0] != "old":
-                    date[3] = Event.search_count(
-                        dom_without({"date_end", "date_begin"}) + date[2]
-                    )
```
Comment (yajo) [inline website_event_filter_selector/controllers/main.py]: I think that, by removing these lines, you're introducing a bug.

At least in previous versions, when you filtered by one criteria, other criterias' counters were updated accordingly.

Example:

1. I have 2 Trainings in Spain and 2 trainings in France.
2. "All types" displays count 4️⃣ 
3. I filter by country=France
4. "All types" displays count 2️⃣ 

The same should apply to the cities filter here.

File: website_event_filter_selector/readme/CONTRIBUTORS.rst [hunk 2]
```diff
@@ -0,0 +1,9 @@
+* Dennis Sluijk <d.sluijk@onestein.nl>
+* Cristina Martin R.
+
+* `Tecnativa <https://www.tecnativa.com>`_:
+
+    * Rafael Blasco
+    * Pedro Baeza
+    * Jairo Llopiso
```
Comment (yajo) [inline website_event_filter_selector/readme/CONTRIBUTORS.rst]:
```suggestion
    * Jairo Llopis
```

**Review by yajo — CHANGES_REQUESTED** [review]

File: website_event_filter_selector/readme/CONTRIBUTORS.rst (hunk 2 above)
Comment (victoralmau) [reply website_event_filter_selector/readme/CONTRIBUTORS.rst]: Sorry!!

**Comment by victoralmau** [comment]
Addon changed to website_event_filter_city.
Screenshot  added.
Improve controller to keep domain search and list only cities related to events listed.

**Review by yajo — CHANGES_REQUESTED** [review]
I'm afraid the problem from https://github.com/OCA/event/pull/208#discussion_r589239929 still happens.

Just check this out:

imagen

imagen

I understand that you were trying to clean up the controller code, and I agree it was very ugly, but if it worked fine and now it doesn't... better leave it as it was 😅 

The point you have to keep in mind is that the counters and options of *all* filters depend on what is currently being displayed.

For example, if I have no filters, and "Next events" are selected, the categories dropdown filters only next events:

imagen

But if I choose "This week", the categories dropdown filters events for just this week:

imagen

That's how all upstream filters work, and that's how the cities one must work too.

**Comment by victoralmau** [comment]
Ok, thanks for the explanation, if I have understood everything correctly it is already solved.

**Comment by yajo** [comment]
It doesn't seem so:

imagen

File: website_event_filter_city/tests/test_ui.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2016 Tecnativa - Jairo Llopis
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo.tests.common import HttpCase
```
Comment (joao-p-marques) [inline website_event_filter_city/tests/test_ui.py]:
```suggestion
from odoo.tests.common import HttpCase, tagged
```

File: website_event_filter_city/tests/test_ui.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+
```
Comment (joao-p-marques) [inline website_event_filter_city/tests/test_ui.py:6]:
```suggestion
@tagged("post_install", "-at_install")
```

File: website_event_filter_city/tests/test_ui.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    def test_ui_website(self):
+        """Test frontend tour."""
+        self.phantom_js(
+            url_path="/event",
+            code="odoo.__DEBUG__.services['web_tour.tour']"
+            ".run('website_event_filter_city')",
+            ready="odoo.__DEBUG__.services['web_tour.tour']"
+            ".tours.website_event_filter_city.ready",
+            login="demo",
+        )
```
Comment (joao-p-marques) [inline website_event_filter_city/tests/test_ui.py]: You could simplify this:
```suggestion
        self.start_tour("/event", "website_event_filter_city", login="portal")
```
Also, adding the `login="portal"` avoids the 404 error you are getting.

**Review by joao-p-marques — CHANGES_REQUESTED** [review]

**Comment by victoralmau** [comment]
Tour is already fixed.
Can you check it again?

File: website_event_filter_city/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2016-2017 Tecnativa - Jairo Llopis
+# Copyright 2019 Tecnativa - Cristina Martin R.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+{
+    "name": "Website Event Filter City",
+    "summary": "Add a customizable top area to filter events with city",
+    "version": "13.0.1.2.0",
```
Comment (joao-p-marques) [inline website_event_filter_city/__manifest__.py]:
```suggestion
    "version": "13.0.1.0.0",
```

**Review by joao-p-marques — CHANGES_REQUESTED** [review]

**Review by yajo — APPROVED** [review]
Code looks perfect now, and functional tests passed. Congratulations, this was a hard one. ✊🏼 🎉

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**